server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -533,11 +547,12 @@ private void reroute(RoutingAllocation allocation) {
         assert AutoExpandReplicas.getAutoExpandReplicaChanges(allocation.metadata(), allocation).isEmpty()
             : "auto-expand replicas out of sync with number of nodes in the cluster";
         assert assertInitialized();
-
+        long rerouteStartTimeMS = System.currentTimeMillis();
         removeDelayMarkers(allocation);
 
         allocateExistingUnassignedShards(allocation);  // try to allocate existing shard copies first
         shardsAllocator.allocate(allocation);
+        this.rerouteHistogram.record((double) Math.max(0, System.currentTimeMillis() - rerouteStartTimeMS));
```

Comment:
To measure latencies it's advised to use System::nanoTime.

Reply:
Ack, just to clarify:
Should we use ```System::nanoTime``` and then convert to ```ms``` and record or are you suggesting to record the latencies in nanoSeconds itself?

Reply:
You can use something like this - https://github.com/opensearch-project/OpenSearch/blob/18e58160a70595ed450c663b78445abb795499d3/server/src/main/java/org/opensearch/indices/IndicesService.java#L1620

Reply:
Thanks for sharing the example. Updated with the same.